File: Niveau 4/4 – Récursivité avancée/1 - Changement de nom.py
def backtrack(lettres, size, curr):
   if (len(curr) == size):
      print(curr)
      return
   
   for c in lettres:
      curr += c
      backtrack(lettres, size, curr)
      curr = curr[:-1]
      
def main():
   nbLettre = int(input())
   lettres = input()
   size = int(input())
   
   print(nbLettre ** size)
   
   curr = ""
   backtrack(lettres, size, curr)
   
main()

File: Niveau 4/4 – Récursivité avancée/1 - Changement de nom.cpp
#include <cmath>
#include <iostream>
#include <vector>

using namespace std;

void backtrack(vector<char> &letters, int size, string &curr)
{
    if (curr.size() == size)
    {
        cout << curr << '\n';
        return;
    }

    for (char c : letters)
    {
        curr.push_back(c);
        backtrack(letters, size, curr);
        curr.pop_back();
    }
}

int main()
{
    int nbLettre;
    cin >> nbLettre;

    vector<char> arr(nbLettre, ' ');

    for (int i = 0; i < nbLettre; i++)
    {
        cin >> arr[i];
    }

    int size;
    cin >> size;

    cout << pow(nbLettre, size) << '\n';

    string curr;
    backtrack(arr, size, curr);
}